include/miopen/miopen.h

```diff
@@ -5730,6 +5730,536 @@ MIOPEN_EXPORT miopenStatus_t miopenGroupNormForward(miopenHandle_t handle,
+ * void *arrayOfElements argument of miopenBackendSetAttribute()
+ * and miopenBackendGetAttribute()
+ */
+typedef enum
+{
+    MIOPEN_TYPE_HANDLE = 0,              /*!< miopenHandle_t */
+    MIOPEN_TYPE_DATA_TYPE,               /*!< miopenDataType_t */
+    MIOPEN_TYPE_BOOLEAN,                 /*!< bool */
+    MIOPEN_TYPE_INT64,                   /*!< int64_t */
+    MIOPEN_TYPE_FLOAT,                   /*!< float */
+    MIOPEN_TYPE_DOUBLE,                  /*!< double */
+    MIOPEN_TYPE_VOID_PTR,                /*!< void * */
+    MIOPEN_TYPE_CONVOLUTION_MODE,        /*!< miopenConvolutionMode_t */
+    MIOPEN_TYPE_HEUR_MODE,               /*!< miopenBackendHeurMode_t */
+    MIOPEN_TYPE_KNOB_TYPE,               /*!< miopenBackendKnobType_t */
+    MIOPEN_TYPE_NAN_PROPOGATION,         /*!< miopenNanPropagation_t */
+    MIOPEN_TYPE_NUMERICAL_NOTE,          /*!< miopenBackendNumericalNote_t */
+    MIOPEN_TYPE_LAYOUT_TYPE,             /*!< miopenBackendLayoutType_t */
+    MIOPEN_TYPE_ATTRIB_NAME,             /*!< miopenBackendAttributeName_t */
+    MIOPEN_TYPE_POINTWISE_MODE,          /*!< miopenPointwiseMode_t */
+    MIOPEN_TYPE_BACKEND_DESCRIPTOR,      /*!< miopenBackendDescriptor_t */
+    MIOPEN_TYPE_GENSTATS_MODE,           /*!< miopenGenStatsMode_t */
+    MIOPEN_TYPE_BN_FINALIZE_STATS_MODE,  /*!< miopenBnFinalizeStatsMode_t */
+    MIOPEN_TYPE_REDUCTION_OPERATOR_TYPE, /*!< miopenReduceTensorOp_t */
+    MIOPEN_TYPE_BEHAVIOR_NOTE,           /*!< miopenBackendBehaviorNote_t */
+    MIOPEN_TYPE_TENSOR_REORDERING_MODE,  /*!< miopenBackendTensorReordering_t */
+    MIOPEN_TYPE_RESAMPLE_MODE,           /*!< miopenResampleMode_t */
+    MIOPEN_TYPE_PADDING_MODE,            /*!< miopenPaddingMode_t */
+    MIOPEN_TYPE_INT32,                   /*!< int32_t */
+    MIOPEN_TYPE_CHAR,                    /*!< char */
+    MIOPEN_TYPE_SIGNAL_MODE,             /*!< miopenSignalMode_t */
+    MIOPEN_TYPE_FRACTION,                /*!< miopenFraction_t */
+    MIOPEN_TYPE_NORM_MODE,               /*!< miopenBackendNormMode_t */
+    MIOPEN_TYPE_NORM_FWD_PHASE,          /*!< miopenBackendNormFwdPhase_t */
+    MIOPEN_TYPE_RNG_DISTRIBUTION         /*!< miopenRngDistribution_t */
+} miopenBackendAttributeType_t;
+
+/*! @brief Backend descriptor
+ *
+ * A typedef void pointer to one of many opaque descriptor structures.
+ * The type of structure that it points to is determined by the argument when allocating the memory
+ * for the opaque structure using miopenBackendCreateDescriptor().
+ *
+ * Attributes of a descriptor can be set using miopenBackendSetAttribute(). After all required
+ * attributes of a descriptor are set, the descriptor can be finalized by miopenBackendFinalize().
+ * From a finalized descriptor, one can query its queryable attributes using
+ * miopenBackendGetAttribute(). Finally, the memory allocated for a descriptor can be freed using
+ * miopenBackendDestroyDescriptor().
+ */
+typedef void* miopenBackendDescriptor_t;
```

Comment:
Use `MIOPEN_DECLARE_OBJECT(miopenBackendDescriptor)` and that defines an empty base struct `miopenBackendDescriptor_t`. You can see examples in this file and the definition too.